gazel/gazel.go

```diff
@@ -49,9 +53,17 @@ func main() {
 	if err := v.walkRepo(); err != nil {
 		glog.Fatalf("err walking repo: %v", err)
 	}
-	if _, err := v.reconcileAllRules(); err != nil {
+	written := 0
+	if written, err = v.reconcileAllRules(); err != nil {
 		glog.Fatalf("err reconciling rules: %v", err)
 	}
+	if *validate {
+		if written > 0 {
+			fmt.Fprintf(os.Stderr, "\n%d BUILD files not up-to-date.\n", written)
+			os.Exit(1)
+		}
+		fmt.Fprintf(os.Stderr, "All BUILD files up-to-date.\n")
```

Comment:
I kinda like having the affirmation that everything is OK, but I can remove this.